server/mqtt.go

```diff
@@ -382,7 +400,7 @@ type mqttReader struct {
 }
 
 type mqttWriter struct {
-	bytes.Buffer
+	*bytes.Buffer
```

Comment:
It's an extra indirection, but makes initializing it so much easier. Consider avoiding the extra pointer.